librz/asm/arch/evm/evmasm.c

```diff
@@ -0,0 +1,359 @@
+	} else if (!strcmp("iszero", buf)) {
+		opbuf[0] = 0x15;
+	} else if (!strcmp("and", buf)) {
+		opbuf[0] = 0x16;
+	} else if (!strcmp("or", buf)) {
+		opbuf[0] = 0x17;
+	} else if (!strcmp("xor", buf)) {
+		opbuf[0] = 0x18;
+	} else if (!strcmp("not", buf)) {
+		opbuf[0] = 0x19;
+	} else if (!strcmp("byte", buf)) {
+		opbuf[0] = 0x1a;
+	} else if (!strcmp("shl", buf)) {
+		opbuf[0] = 0x1b;
+	} else if (!strcmp("shr", buf)) {
+		opbuf[0] = 0x1c;
+	} else if (!strcmp("sar", buf)) {
+		opbuf[0] = 0x1d;
+	} else if (!strcmp("sha3", buf)) {
+		opbuf[0] = 0x20;
+	} else if (!strcmp("address", buf)) {
+		opbuf[0] = 0x30;
+	} else if (!strcmp("balance", buf)) {
+		opbuf[0] = 0x31;
+	} else if (!strcmp("origin", buf)) {
+		opbuf[0] = 0x32;
+	} else if (!strcmp("caller", buf)) {
+		opbuf[0] = 0x33;
+	} else if (!strcmp("callvalue", buf)) {
+		opbuf[0] = 0x34;
+	} else if (!strcmp("calldataload", buf)) {
+		opbuf[0] = 0x35;
+	} else if (!strcmp("calldatasize", buf)) {
+		opbuf[0] = 0x36;
+	} else if (!strcmp("calldatacopy", buf)) {
+		opbuf[0] = 0x37;
+	} else if (!strcmp("codesize", buf)) {
+		opbuf[0] = 0x38;
+	} else if (!strcmp("codecopy", buf)) {
+		opbuf[0] = 0x39;
+	} else if (!strcmp("gasprice", buf)) {
+		opbuf[0] = 0x3a;
+	} else if (!strcmp("extcodesize", buf)) {
+		opbuf[0] = 0x3b;
+	} else if (!strcmp("extcodecopy", buf)) {
+		opbuf[0] = 0x3c;
+	} else if (!strcmp("returndatasize", buf)) {
+		opbuf[0] = 0x3d;
+	} else if (!strcmp("returndatacopy", buf)) {
+		opbuf[0] = 0x3e;
+	} else if (!strcmp("extcodehash", buf)) {
+		opbuf[0] = 0x3f;
+	} else if (!strcmp("blockhash", buf)) {
+		opbuf[0] = 0x40;
+	} else if (!strcmp("coinbase", buf)) {
+		opbuf[0] = 0x41;
+	} else if (!strcmp("timestamp", buf)) {
+		opbuf[0] = 0x42;
+	} else if (!strcmp("number", buf)) {
+		opbuf[0] = 0x43;
+	} else if (!strcmp("difficulty", buf)) {
+		opbuf[0] = 0x44;
+	} else if (!strcmp("gaslimit", buf)) {
+		opbuf[0] = 0x45;
+	} else if (!strcmp("chainid", buf)) {
+		opbuf[0] = 0x46;
+	} else if (!strcmp("selfbalance", buf)) {
+		opbuf[0] = 0x47;
+	} else if (!strcmp("pop", buf)) {
+		opbuf[0] = 0x50;
+	} else if (!strcmp("mload", buf)) {
+		opbuf[0] = 0x51;
+	} else if (!strcmp("mstore", buf)) {
+		opbuf[0] = 0x52;
+	} else if (!strcmp("mstore8", buf)) {
+		opbuf[0] = 0x53;
+	} else if (!strcmp("sload", buf)) {
+		opbuf[0] = 0x54;
+	} else if (!strcmp("sstore", buf)) {
+		opbuf[0] = 0x55;
+	} else if (!strcmp("jump", buf)) {
+		opbuf[0] = 0x56;
+	} else if (!strcmp("jump", buf)) {
+		opbuf[0] = 0x56;
+	} else if (!strcmp("jumpi", buf)) {
+		opbuf[0] = 0x57;
+	} else if (!strcmp("pc", buf)) {
+		opbuf[0] = 0x58;
+	} else if (!strcmp("msize", buf)) {
+		opbuf[0] = 0x59;
+	} else if (!strcmp("gas", buf)) {
+		opbuf[0] = 0x5a;
+	} else if (!strcmp("jumpdest", buf)) {
+		opbuf[0] = 0x5b;
+	} else if (!strncmp("push", buf, 3)) {
+		char out[100];
+		int number = atoi(buf+4);
+		switch (number) {
+			case 1:
+			len = 2;
+			opbuf[0] = 0x60;
+			break;
+		case 2:
```

Comment:
you can optimize this out by using a structure like suggested before which contains name, length, and the first byte.